WordPress/WordPressTest/TestUtilities/JSONLoader.swift

```diff
@@ -38,6 +34,17 @@ import Foundation
         return nil
     }
 
+    /// Loads the specified json file and returns a dictionary representing it.
+    ///
+    /// - Parameter filename: The full name of the json file to load.
+    /// - Returns: A dictionary representing the contents of the json file.
+    open func loadFile(named filename: String) -> JSONDictionary? {
```

Comment:
Agreed. I don't think the `JSONLoader` instance is useful either. We could also add the JSON parsing function as an extension method to `JSONDictionary`, to save us from creating another arbitrary class `JSONLoader`, which isn't used in many places.

Reply:
If we're introducing a new static method anyway, I wonder if we could adjust the signature while we're at it?
```suggestion
    open func loadFile(named filename: String) throws -> JSONDictionary {
```

`XCTest` knows what to do with `throw`-ing functions, and the various `try?` calls here are swallowing errors that could make debugging easier 😅

WDYT?

Reply:
Yes definitely. As I mentioned in [this comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/18478#discussion_r862666547) below, making this function throw was my initial thought, but the changes involved in various call sites made me go with the change that's more "compatible" with existing code.

But I went [a step further](https://github.com/wordpress-mobile/WordPress-iOS/pull/18478#discussion_r863237278) to introduce a new pattern to load fixtures from the unit test bundle, I would appreciate your feedback on the new approach.

Reply:
Thank @jkmassel! I don't know why, but I never reach for `throw` when thinking about API design in Swift, yet it's such a recommended and well-supported language-wise approach 😳 

As you mention, XCTest the framework has built-in handling for tests that throw at some point in the chain, so that sounds like a great approach. It would also be inline with what we do [in `ScreenObject`](https://github.com/Automattic/ScreenObject/blob/2d65beae47cdcaaf21703ce1b321f382fe0d0730/Sources/ScreenObject/ScreenObject.swift#L71) for the UI tests.

@crazytonyli [mentioned](https://github.com/wordpress-mobile/WordPress-iOS/pull/18478#discussion_r862666547) the extra work involved in updating all the call stack to handle the `throws` and I can see how that might be daunting and would also result in a noisy PR diff. A suggestion that came to mind is to wrap this PR with a `fatalError` then follow up with one that replaces it with the throwing behavior.